### PR TITLE
fix: EV can have low fuel light

### PR DESCRIPTION
**Pull request by dahlb**
some EV have vehicleStatus.lowFuelLight like the Niro EV 2019

Am I missing something? why not just put them all in the array? They only populate if the api response has a value, is there a use case where the api has a value and it is better to hide the sensor?

Screen Shot 2021-11-13 at 8 08 27 PM

### CODE REVIEW

**Comment by cdnninja** [comment]
I think this was a outcome from improvements.  At first the logic didn't exist to check for data.  So when the Canadian side was built I had lots of empty sensors.  I added that check to walk through them all. I would think we should just move them all into the array.

**Comment by cdnninja** [comment]
Would it be worth in this case though having a different sensor added to the EV logic?  Could label it low EV battery instead of fuel.

File: custom_components/kia_uvo/binary_sensor.py [hunk 1]
```diff
@@ -37,15 +37,12 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         ("backWindowHeater", "Back Window Heater", "vehicleStatus.sideBackWindowHeat", "mdi:car-defrost-rear", "mdi:car-defrost-rear", None),
         ("sideMirrorHeater", "Side Mirror Heater", "vehicleStatus.sideMirrorHeat", "mdi:car-side", "mdi:car-side", None),
         ("steeringWheelHeater", "Steering Wheel Heater", "vehicleStatus.steerWheelHeat", "mdi:steering", "mdi:steering", None),
+        ("lowFuelLight", "Low Fuel Light", "vehicleStatus.lowFuelLight", "mdi:gas-station-off", "mdi:gas-station", None),
```
Comment (fuatakgun) [inline custom_components/kia_uvo/binary_sensor.py:40]: are you sure, is this stored in same attribute? `lowFuelLight`?

Comment (dahlb) [reply custom_components/kia_uvo/binary_sensor.py:40]: yes @fuatakgun

this is the response for my EV
```
{"status":{"statusCode":0,"errorType":0,"errorCode":0,"errorMessage":"Success with response body"},"payload":{"vehicleInfoList":[{"vinKey":"kkk","vehicleConfig":{"vehicleDetail":{"vehicle":{"vin":"vvv","trim":{"modelYear":"2019","salesModelCode":"V1262","optionGroupCode":"015","modelName":"NIRO EV","factoryCode":"DQ","projectCode":"DEEV","trimName":"EX PREMIUM","driveType":"0","transmissionType":"1","ivrCategory":"6","btSeriesCode":"N"},"telematics":1,"mileage":"12292.7","mileageSyncDate":"20211115003822","exteriorColor":"ALUMINUM SILVER","exteriorColorCode":"C3S","fuelType":4,"invDealerCode":"MD047","testVehicle":"0","supportedApps":[{"appType":"0"},{"appType":"5","appImage":{"imageName":"uvo-app.png","imagePath":"/content/dam/kia/us/owners/image/common/app/access/","imageType":"2","imageSize":{"length":"100","width":"100","uom":0}}}],"activationType":2},"images":[{"imageName":"2019-niro_ev-ex_premium-c3s.png","imagePath":"/content/dam/kia/us/owners/image/vehicle/2019/niro_ev/ex_premium/","imageType":"1","imageSize":{"length":"100","width":"100","uom":0}}],"device":{"launchType":"0","swVersion":"DEEV.USA.SOP.V105.190503.STD_H","telematics":{"generation":"3","platform":"1","tmsCenter":"1","billing":true},"versionNum":"ECO","headUnitType":"0","hdRadio":"X40HA","ampType":"NA","modem":{"meid":"354522081158920","mdn":"6573652075","iccid":"89148000005051202869"},"headUnitName":"avn40ev_np","bluetoothRef":"10","headUnitDesc":"AVN5.0"}},"billingPeriod":{"freeTrial":{"value":12,"unit":0},"freeTrialExtension":{"value":12,"unit":1},"servicePeriod":{"value":60,"unit":1}}},"lastVehicleInfo":{"vehicleNickName":"Niro EV","preferredDealer":"MD047","customerType":0,"enrollment":{"provStatus":"4","enrollmentStatus":"1","enrollmentType":"0","registrationDate":"20191123","expirationDate":"20201122","expirationMileage":"100000","freeServiceDate":{"startDate":"20191122","endDate":"20201122"}},"activeDTC":{"dtcActiveCount":"0"},"vehicleStatusRpt":{"statusType":"2","reportDate":{"utc":"20211115164556","offset":-8},"vehicleStatus":{"climate":{"airCtrl":false,"defrost":false,"airTemp":{"value":"72","unit":1},"heatingAccessory":{"steeringWheel":0,"sideMirror":0,"rearWindow":0}},"engine":false,"doorLock":false,"doorStatus":{"frontLeft":0,"frontRight":0,"backLeft":0,"backRight":0,"trunk":0,"hood":0},"lowFuelLight":false,"evStatus":{"batteryCharge":false,"batteryStatus":84,"batteryPlugin":0,"remainChargeTime":[{"remainChargeType":1,"timeInterval":{"value":0,"unit":4}},{"remainChargeType":2,"timeInterval":{"value":0,"unit":4}},{"remainChargeType":3,"timeInterval":{"value":80,"unit":4}}],"drvDistance":[{"type":2,"rangeByFuel":{"evModeRange":{"value":226,"unit":3},"totalAvailableRange":{"value":226,"unit":3}}}],"syncDate":{"utc":"20211115153822","offset":-8},"targetSOC":[{"plugType":0,"targetSOClevel":127,"dte":{"type":2,"rangeByFuel":{"gasModeRange":{"value":0,"unit":3},"evModeRange":{"value":226,"unit":3},"totalAvailableRange":{"value":226,"unit":3}}}},{"plugType":1,"targetSOClevel":127,"dte":{"type":2,"rangeByFuel":{"gasModeRange":{"value":0,"unit":3},"evModeRange":{"value":226,"unit":3},"totalAvailableRange":{"value":226,"unit":3}}}}]},"ign3":false,"transCond":true,"tirePressure":{"all":0},"dateTime":{"utc":"20211115164556","offset":-8},"syncDate":{"utc":"20211115153822","offset":-8},"batteryStatus":{"stateOfCharge":90,"sensorStatus":0},"sleepMode":true,"lampWireStatus":{"headLamp":{},"stopLamp":{},"turnSignalLamp":{}},"windowStatus":{},"engineRuntime":{},"valetParkingMode":0}},"location":{"coord":{"lat":4,"lon":54,"alt":106,"type":0,"altdo":0},"head":345,"speed":{"value":0,"unit":1},"accuracy":{"hdop":6,"pdop":12},"syncDate":{"utc":"20211115153822","offset":-8}},"financed":true,"financeRegistered":true,"linkStatus":0}}]}}
```

**Comment by dahlb** [comment]
I could move the others into the array too if they are just an artifact of before it was safe to have the array contain attributes that don't exist. thoughts?

**Comment by dahlb** [comment]
I'd prefer less branching since it uses the same attribute name, but I'm open to using an if with battery-arrow-up-outline
icon for EV and the fual one for others if @cdnninja or @fuatakgun have a preference

**Comment by cdnninja** [comment]
No preference from me.  I am good eitherway.

**Comment by fuatakgun** [comment]
i am also good to share same sensor between ev and non-ev vehicles

**Pull request merged**